app/src/main/java/samyups/example/trainingss/ui/MainActivity.kt

```diff
@@ -0,0 +1,47 @@
+package samyups.example.trainingss.ui
+
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import androidx.lifecycle.Observer
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.LinearLayoutManager
+import kotlinx.android.synthetic.main.activity_main.*
+import kotlinx.android.synthetic.main.add_dogs_dialog_box.*
+import kotlinx.android.synthetic.main.add_dogs_dialog_box.view.*
+import samyups.example.trainingss.*
+import samyups.example.trainingss.util.InjectorUtils
+
+class MainActivity : AppCompatActivity() {
+
+    private val mainAdapter = MainRecyclerAdapter()
+    private lateinit var mainDialogBox : MainDialogBox
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        initRecyclerView()
+        initUI(mainAdapter)
```

Comment:
You don't have to pass `mainAdapter ` as a function argument here. `mainAdapter` is a global variable, or an instance variable. It's already available to all the methods in this class. You can just access it directly in `initUI` without having to pass it via the method signature.